```
  ======================================================================
  ERROR: corehq.apps.case_search.tests.test_ancestor_functions:TestAncestorQueries.test_host
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "/mnt/commcare-hq-ro/corehq/apps/case_search/tests/test_ancestor_functions.py", line 127, in test_host
      query1 = get_case_search_query(
    File "/mnt/commcare-hq-ro/corehq/apps/case_search/tests/utils.py", line 12, in get_case_search_query
      return builder.build_query(criteria, commcare_sort=commcare_sort)
    File "/mnt/commcare-hq-ro/corehq/apps/case_search/utils.py", line 249, in build_query
      search_es = self._apply_filter(search_es, criteria)
    File "/mnt/commcare-hq-ro/corehq/apps/case_search/utils.py", line 281, in _apply_filter
      search_es = search_es.filter(self._build_filter_from_xpath(xpath))
    File "/mnt/commcare-hq-ro/corehq/apps/case_search/utils.py", line 298, in _build_filter_from_xpath
      return build_filter_from_xpath(self.query_domains, xpath, fuzzy,
    File "/mnt/commcare-hq-ro/corehq/apps/case_search/filter_dsl.py", line 150, in build_filter_from_xpath
      raise CaseFilterError(_("Malformed search query"), None)
  corehq.apps.case_search.exceptions.CaseFilterError: Malformed search query
```

--- a/corehq/apps/case_search/xpath_functions/ancestor_functions.py
+++ b/corehq/apps/case_search/xpath_functions/ancestor_functions.py
@@ -10,6 +10,17 @@ from corehq.apps.es.case_search import CaseSearchES, reverse_index_case_query
 from corehq.toggles import NO_SCROLL_IN_CASE_SEARCH
 
 
+def _should_scroll(context):
+    if not isinstance(context.domain, list):
+        domains = [context.domain]
+    else:
+        domains = context.domain
+    for domain in domains:
+        if NO_SCROLL_IN_CASE_SEARCH.enabled(domain):
+            return False
+    return True
+
+
 def is_ancestor_comparison(node):
     """Returns whether a particular AST node is an ancestor comparison
 
@@ -92,10 +103,10 @@ def _child_case_lookup(context, case_ids, identifier):
     """
     es_query = CaseSearchES().domain(context.domain).get_child_cases(case_ids, identifier)
     context.profiler.add_query('_child_case_lookup', es_query)
-    if NO_SCROLL_IN_CASE_SEARCH.enabled(context.domain):
-        return es_query.get_ids()
-    else:
+    if _should_scroll(context.domain):
         return es_query.scroll_ids()
+    else:
+        return es_query.get_ids()
 
 
 def ancestor_exists(node, context):
@@ -158,7 +169,7 @@ def _get_case_ids_from_ast_filter(context, filter_node):
                 new_query
             )
 
-        if NO_SCROLL_IN_CASE_SEARCH.enabled(context.domain):
-            return es_query.get_ids()
-        else:
+        if _should_scroll(context.domain):
             return es_query.scroll_ids()
+        else:
+            return es_query.get_ids()

--- a/corehq/apps/case_search/xpath_functions/ancestor_functions.py
+++ b/corehq/apps/case_search/xpath_functions/ancestor_functions.py
@@ -103,7 +103,7 @@ def _child_case_lookup(context, case_ids, identifier):
     """
     es_query = CaseSearchES().domain(context.domain).get_child_cases(case_ids, identifier)
     context.profiler.add_query('_child_case_lookup', es_query)
-    if _should_scroll(context.domain):
+    if _should_scroll(context):
         return es_query.scroll_ids()
     else:
         return es_query.get_ids()
@@ -169,7 +169,7 @@ def _get_case_ids_from_ast_filter(context, filter_node):
                 new_query
             )
 
-        if _should_scroll(context.domain):
+        if _should_scroll(context):
             return es_query.scroll_ids()
         else:
             return es_query.get_ids()

